### PR TITLE
Apply default Mapstory style to imports

### PR DESCRIPTION
**Pull request by goudine**
Applies a default MapStory SLD style based on the feature type of imported layer.

### CODE REVIEW

File: mapstory/import_handlers.py [hunk 1]
```diff
@@ -51,3 +53,63 @@ def handle(self, layer, layer_config, *args, **kwargs):
                 for date_option in ('start_date', 'end_date'):
                     if layer_config.get(date_option) == date_field:
                         layer_config[date_option] = new_field_name.lower()
+
+
+class GeoserverHandlerMixin(ImportHandlerMixin):
+    """
+    A Mixin for Geoserver handlers.
```
Comment (DBlasby) [inline mapstory/import_handlers.py]: please make more specific

File: mapstory/import_handlers.py [hunk 2]
```diff
@@ -51,3 +53,63 @@ def handle(self, layer, layer_config, *args, **kwargs):
...
+            return False
+
+    @ensure_can_run
+    def handle(self, layer, layer_config, *args, **kwargs):
+        """
+        Default MapStory style applied depending on feature type
+        """
+
+        # import pydevd
+        # pydevd.settrace('192.168.0.15', port=65432, stdoutToServer=True, stderrToServer=True, trace_only_current_thread=False,overwrite_prev_trace=True,patch_multiprocessing=True)
```
Comment (DBlasby) [inline mapstory/import_handlers.py]: remove this

File: mapstory/import_handlers.py [hunk 3]
```diff
@@ -51,3 +53,63 @@ def handle(self, layer, layer_config, *args, **kwargs):
...
+
+        sld = self.get_feature_type(self.layer)
+
+        layer = self.catalog.get_layer(self.layer.name)
```
Comment (DBlasby) [inline mapstory/import_handlers.py]: layer is already included as an arg to this method

File: mapstory/import_handlers.py [hunk 4]
```diff
@@ -51,3 +53,63 @@ def handle(self, layer, layer_config, *args, **kwargs):
...
+            for attr in ft.dom.find("attributes").getchildren():
+                attr_binding = attr.find("binding")
+                if "jts.geom" in attr_binding.text:
+                    if "Polygon" in attr_binding.text:
+                        name = "Generic_Mapstory_Polygon"
+                    elif "Line" in attr_binding.text:
+                        name = "Generic_Mapstory_Line"
+                    else:
+                        name = "Generic_Mapstory_Point"
+        return name
```
Comment (DBlasby) [inline mapstory/import_handlers.py]: might want to return "generic" as the default (if it cannot determine which one it is).
"generic" is a default GS style that should render any layer (but a wee bit inefficiency).

File: mapstory/settings/base.py [hunk 5]
```diff
@@ -291,7 +291,7 @@ def str_to_bool(v):
     'osgeo_importer.handlers.geoserver.GeoServerTimeHandler',
     'osgeo_importer.handlers.geoserver.GeoWebCacheHandler',
     'osgeo_importer.handlers.geonode.GeoNodePublishHandler',
-    'osgeo_importer.handlers.geoserver.GenericSLDHandler',
+    'mapstory.import_handlers.DefaultStyleHandler',
```
Comment (DBlasby) [inline mapstory/settings/base.py]: I think this SLD handler needs to happen before the GeonodePublish handler (or thumbnails will be created before the style is applied)

**Review by DBlasby — COMMENTED** [review]
just a few minor points...

File: mapstory/import_handlers.py (hunk 2 above)
Comment (goudine) [reply mapstory/import_handlers.py]: removed, thanks for catching this

File: mapstory/settings/base.py (hunk 5 above)
Comment (goudine) [reply mapstory/settings/base.py]: order switched

File: mapstory/import_handlers.py (hunk 3 above)
Comment (goudine) [reply mapstory/import_handlers.py]: renamed variable to make more clear

File: mapstory/import_handlers.py (hunk 4 above)
Comment (goudine) [reply mapstory/import_handlers.py]: added generic style as 'else' case

**Comment by cuttlefish** [comment]
Is this PR supposed to be against master?

**Comment by DBlasby** [comment]
likely best to push against master and merge into this branch.

**Comment by cuttlefish** [comment]
Also, setting up the PR against non-master has bypassed the Travis build and review requirements.

**Comment by goudine** [comment]
@cuttlefish I didn't want to make PR against `master` because there was some of your commits in there and I figure you may way to get those merged in yourself.

I can re-PR against `master` if you think it's best to do so

**Comment by goudine** [comment]
closing this to PR against master instead

**Pull request closed without merging**